docs/upgrade_guide.md

```diff
@@ -0,0 +1,46 @@
+# Upgrade Guide
+
+This guide describes how to upgrade the Open Service Mesh (OSM) control plane.
+
+## Prerequisites
+- Kubernetes cluster with the OSM control plane installed
+- The [helm 3 CLI](https://helm.sh/docs/intro/install/) 
+  - The helm CLI must be used until osm CLI upgrade support is implemented.
+
+## Resource availability during upgrade
+Since upgrades include redeploying the osm-controller with the new version, there may be some downtime of the controller. This means that there will be a delay in processing new SMI resources. 
+
+However, already existing SMI resources will be unaffected (assuming [CRD Upgrades](#CRD-Upgrades) are not needed). This means that the data plane (which includes the Envoy sidecar configs) will also be unaffected by upgrading. 
+
+Data plane interruptions are expected if the upgrade includes CRD changes.
+
+## CRD Upgrades
+Please check the `CRD Updates` section of the [release notes](https://github.com/openservicemesh/osm/releases) to see if additional steps are required to update the CRDs used by OSM. If the new release does contain updates to the CRDs, it is required to first delete existing CRDs and the associated Custom Resources prior to upgrading.
+
+In the `./scripts/cleanup` directory we have included a helper script to delete those CRDs and Custom Resources: `./scripts/cleanup/crd-cleanup.sh`
+
+After upgrading, the Custom Resources will need to be recreated using the updated CRDs installed by the upgrade.
+
+## OSM Configuration
+When upgrading, any custom settings used to install or run OSM may be reverted to the default. This includes any metrics deployments and any changes to the OSM ConfigMap. Please ensure that you carefully follow the guide to prevent these values from being overwritten.
+
+## Upgrade OSM using Helm
+Use the `helm` CLI to upgrade the OSM control plane.
+
+### Preserve OSM configuration
```

Comment:
Have you gotten a chance to play with the `--reuse-values` flag with `helm upgrade` ? It preserves the last release's values and merge in any overrides from the command line via --set and -f.

Making the command:
`helm upgrade osm osm --reuse-values --repo https://openservicemesh.github.io/osm --version <chart version> --namespace <osm namespace>